packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts

```diff
@@ -1189,6 +1458,47 @@ export class AutoScalingGroup extends AutoScalingGroupBase implements
     return this.newInstancesProtectedFromScaleIn === true;
   }
 
+  /**
+   * Allows specify security group connections for instances of this fleet.
+   */
+  public get connections(): ec2.Connections {
+    if (this._connections) {
+      return this._connections;
+    }
+
+    if (this.launchTemplate) {
+      return this.launchTemplate.connections;
+    }
+
+    throw new Error('AutoScalingGroup can only be used as IConnectable if it is not created from an imported Launch Template.');
+  }
+
+  /**
+   * UserData for the instances
+   */
+  public get userData(): ec2.UserData {
+    if (this._userData) {
+      return this._userData;
+    }
+
+    if (this.launchTemplate?.userData) {
+      return this.launchTemplate.userData;
+    }
+
+    throw new Error('The provided launch template does not expose its user data.');
+  }
+
+  /**
+   * Role for the instances
+   */
```

Comment:
Can we update this comment to explain what this `Role` is used for? eg if it's an execution role, that should be mentioned here (also, this variable probably needs a better name than `role`; perhaps `executionRole` or `instanceExecutionRole`, if that's what it represents).

Reply:
I cannot change the getter name here (same as `userData`) because it was a class property. `role` and `userData` are turned into getters so that they can throw exceptions when a Launch Template reference is given.